### PR TITLE
 Update validators for E2E runtime metrics

### PR DESCRIPTION
**Pull request by bjrara**
*Issue description:*
Add E2E canaries for runtime metrics in Java and Python across platforms.

*Description of changes:*
1. Code refactoring CWMetricValidator, allow verification with `ANY_VALUE` metric dimension value.
2. Update log query excluding Operation.

*Rollback procedure:*

<Can we safely revert this commit if needed? If not, detail what must be done to safely revert and why it is needed.>

*Ensure you've run the following tests on your changes and include the link below:*

* Java: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/11297668229/job/31425009798
* Python: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/11301593419/job/31436219251

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java
Comment (majanjua-amzn) [inline validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java:154]: nit: A comment explaining why we wouldn't need operation during validation would be helpful

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (majanjua-amzn) [inline validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:186]: This will print only one of potentially many missing metrics, right? It also doesn't wait until the `matchAny` check is done, could we collect all the metrics we find missing and list them for better understanding of what exactly went wrong?

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (majanjua-amzn) [inline validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:194]: I wonder if there's a way to remove from the actual metric set the same metrics that were removed from `matchExact` during `matchExact.removeAll(actualMetricSet)` since they're already been matched.

This would reduce the number of metrics we iterate through drastically. Not a big issue though since we don't expect a huge number of metrics anyways.

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (majanjua-amzn) [inline validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:252]: out of scope nit: Looks like we have an opportunity to implement RegEx instead of relying on only `ANY_VALUE`. If it's not too much effort the validations could implement some regex and this could apply to all metrics instead of having to split the logic based on any value vs. exact match.

Definitely not a requirement, just a thought

**Review by majanjua-amzn — COMMENTED** [review]
Definitely need to run this against some of our existing canaries/release tests. We can't merge this unless we're 100% sure it will not break our log and metric validation.

Unit tests don't provide enough confidence here, even if they have good coverage, since the real data has many different metric aggregations that can be impacted differently.

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (bjrara) [reply validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:186]: Yes, this is how it works today.

We are able to identify all missing metrics because the deduplicated `actualMetrics` are all printed out. Since we know the template to compare, we can figure out the diff.

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (bjrara) [reply validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:252]: Supporting regex means we can no longer utilize hash set to remove the exact match metrics quickly. It may bring performance concern because the algorithm complexity will increase.

File: validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java
Comment (bjrara) [reply validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java]: Added

File: validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java
Comment (bjrara) [reply validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java:196]: Done.

**Review by harrryr — APPROVED** [review]
LGTM

**Pull request merged**
